packages/ui/react-ui-menu/src/components/Toolbar/ToolbarToggleGroup.tsx

```diff
@@ -2,36 +2,44 @@
 // Copyright 2025 DXOS.org
 //
 
-import React, { useMemo } from 'react';
+import React, { useCallback } from 'react';
 
 import { Toolbar as NaturalToolbar } from '@dxos/react-ui';
 
 import type { ToolbarActionGroupProps } from './defs';
 import { type MenuAction } from '../../defs';
 import { ActionLabel } from '../ActionLabel';
+import { useMenu, useMenuItems } from '../MenuContext';
 
 const ToolbarToggleGroupItem = ({ action }: { action: MenuAction }) => {
-  const { icon, iconOnly, disabled } = action.properties;
+  const { iconSize, onAction } = useMenu();
+  const { icon, iconOnly = true, disabled } = action.properties;
+  const handleClick = useCallback(() => {
+    onAction?.(action);
+  }, [action, onAction]);
   return (
     <NaturalToolbar.ToggleGroupIconItem
       key={action.id}
       value={action.id}
       icon={icon}
+      size={iconSize}
       iconOnly={iconOnly}
       disabled={disabled}
       label={<ActionLabel action={action} />}
+      onClick={handleClick}
+      variant='ghost'
     />
   );
 };
 
-export const ToolbarToggleGroup = ({ actionGroup, graph }: ToolbarActionGroupProps) => {
-  const menuActions = useMemo(() => (graph ? (graph.actions(actionGroup) as MenuAction[]) : []), [actionGroup, graph]);
-  const { selectCardinality, value } = actionGroup.properties;
+export const ToolbarToggleGroup = ({ group, items: propsItems }: ToolbarActionGroupProps) => {
+  const items = useMenuItems(group, propsItems);
+  const { selectCardinality, value } = group.properties;
   return (
-    // TODO(thure): This is extremely difficult to manage, what do?
+    // TODO(thure): The type here is difficult to manage, what do?
     // @ts-ignore
     <NaturalToolbar.ToggleGroup type={selectCardinality} value={value}>
-      {menuActions.map((action) => (
+      {(items as MenuAction[]).map((action) => (
```

Comment:
is this cast because currently only supporting one level of nesting?

Reply:
No, I was wrestling with TS over types being narrower/broader than other types, and sometimes not handling things like separators yet… TS seems to complain in both cases, like if I pass `MenuItem` in a `Node<any>` situation, it complains even though `Node<any>` is broader than `MenuItem`… it’s type debt, which seemed low enough priority for now, but I can leave TODO’s in all such cases if it would help.

Reply:
todo about why it needs to be cast currently would be beneficial I think